lib/net/ssh/transport/cipher_factory.rb

```diff
@@ -94,6 +94,15 @@ def self.get_lengths(name, options = {})
       end
       result
     end
+
+    def self.new_ossl_cipher(ossl_name)
+      if OpenSSL::Cipher.respond_to?(:new)
+        OpenSSL::Cipher.new(ossl_name)
+      else
+        OpenSSL::Cipher::Cipher.new(ossl_name)
+      end
+    end
+    private_class_method :new_ossl_cipher
```

Comment:
It is Ruby 1.9.0 that actually renamed OpenSSL::Cipher::Cipher to OpenSSL::Cipher. I think net-ssh no longer supports Ruby 1.8?